s08compute_wct.py

```diff
@@ -0,0 +1,425 @@
+            min_nonzero=0.25, freqmin=0.1, freqmax=2.0):
+   
+    inx = np.where((freqs >= freqmin) & (freqs <= freqmax))  # Filter frequencies within the specified range
+    dvv, err = np.zeros(len(inx[0])), np.zeros(len(inx[0])) # Initialize dvv and err arrays
+
+    # Weighting function based on WXamp
+    weight_func = np.log(np.abs(WXamp)) / np.log(np.abs(WXamp)).max()
+    zero_idx = np.where((Wcoh < mincoh) | (delta_t > maxdt))
+    wf = (weight_func + abs(np.nanmin(weight_func))) / weight_func.max()
+    wf[zero_idx] = 0
+
+    # Loop through frequency indices for linear regression
+    for ii, ifreq in enumerate(inx[0]):
+        period = 1.0 / freqs[ifreq]
+        lag_max = lag_min + (period * coda_cycles)
+
+        # Coda selection
+        tindex = np.where(((tvec >= -lag_max) & (tvec <= -lag_min)) | ((tvec >= lag_min) & (tvec <= lag_max)))[0]
+
+        if len(tvec) > 2:
+            if not np.any(delta_t[ifreq]):
+                continue
+
+            delta_t[ifreq][tindex] = np.nan_to_num(delta_t[ifreq][tindex])
+            w = wf[ifreq]  # Weighting function for the specific frequency
+            w[~np.isfinite(w)] = 1.0
+
+            # Percentage of non-zero weights
+            nzc_perc = np.count_nonzero(w[tindex]) / len(tindex)
+
+            if nzc_perc >= min_nonzero:
+                m, em = linear_regression(tvec[tindex], delta_t[ifreq][tindex], w[tindex], intercept_origin=True)
+                dvv[ii], err[ii] = -m, em
+            else:
+                dvv[ii], err[ii] = np.nan, np.nan
+        else:
+            print('Not enough points to estimate dv/v for WCT')
+
+    return dvv * 100, err * 100, wf
+
+def xwt(trace_ref, trace_current, fs, ns=3, nt=0.25, vpo=12, freqmin=0.1, freqmax=8.0, nptsfreq=100):
+    """
+    Wavelet coherence transform (WCT).
+
+    The WCT finds regions in time frequency space where the two time
+    series co-vary, but do not necessarily have high power.
+    
+    Modified from https://github.com/Qhig/cross-wavelet-transform
+
+    Parameters
+    ----------
+    trace_ref, trace_current : numpy.ndarray, list
+        Input signals.
+    fs : float
+        Sampling frequency.
+    ns : smoothing parameter. 
+        Default value is 3
+    nt : smoothing parameter. 
+        Default value is 0.25
+    vpo : float,
+        Spacing parameter between discrete scales. Default value is 12.
+        Higher values will result in better scale resolution, but
+        slower calculation and plot.
+        
+    freqmin : float,
+        Smallest frequency
+        Default value is 0.1 Hz
+    freqmax : float,
+        Highest frequency
+        Default value is 8.0 Hz
+    nptsfreq : int,
+        Number of frequency points between freqmin and freqmax.
+        Default value is 100 points
+       
+    Returns
+    """
+    # Choosing a Morlet wavelet with a central frequency w0 = 6
+    mother = wavelet.Morlet(6.)
```

Comment:
would that be a parameter to be configured by users ? (have zero idea what it does). 
Is Morlet the only choice too ?

Reply:
Yes, it could be. There are four different mother wavelet classes (Morlet, Paul, DOG, MexicanHat), and they have their own default values. I could add parameters wavelet_type and wavelet_scale in the config